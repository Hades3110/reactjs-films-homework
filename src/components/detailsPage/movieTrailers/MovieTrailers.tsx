import { useEffect, useState } from 'react'
import ReactPlayer from 'react-player'
import { getFilmVideo } from '../../../services/filmService'
import styles from './movieTrailers.module.scss'

const MovieTrailers = ({ id }: { id: number }) => {
    const [keys, setKeys] = useState<string[]>([])

    useEffect(() => {
        if (id) getFilmVideo(id).then(res => setKeys([res.results[0].key, res.results[1].key]))
    }, [id])


    return (<div className={styles.treiler}>
        {keys.length > 0 ? keys.map((el) => (<div key={el}>
            <ReactPlayer
                className={styles.player}
                url={`https://www.youtube.com/watch?v=${el}`}
                controls={true}
                width={'100%'}
                height={506}
            />
        </div>)) : 'No treiler'}
    </div>)
}

export default MovieTrailers