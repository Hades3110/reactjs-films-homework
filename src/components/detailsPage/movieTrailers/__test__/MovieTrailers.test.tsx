import { create } from "react-test-renderer";
import MovieTrailers from '../MovieTrailers';

describe('MovieTrailers Component', () => {

  it('MovieTrailers snapshot test', () => {
    const component = create(<MovieTrailers id={0} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});