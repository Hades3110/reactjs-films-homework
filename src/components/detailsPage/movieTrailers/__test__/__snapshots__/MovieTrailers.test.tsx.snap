// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTrailers Component MovieTrailers snapshot test 1`] = `
<div
  className="trailer"
>
  No trailer
</div>
`;
